lib/services/database/database_service.dart

```diff
@@ -0,0 +1,239 @@
+import 'dart:async';
+
+import "package:collection/collection.dart";
+import 'package:mobileprism/services/database/album_data_entry.dart';
+import 'package:mobileprism/services/database/cross_table_entry.dart';
+import 'package:mobileprism/services/database/photo_data_entry.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+const String photoDataTableName = "photo_data";
+const String albumDataTableName = "album_data";
+const String keyCrosstableName = "key_crosstable";
+const String photoTableCreationStrg = '''
+      CREATE TABLE IF NOT EXISTS $photoDataTableName(uid TEXT PRIMARY KEY, 
+      panorama INTEGER, width INTEGER, height INTEGER,
+      image_hash TEXT, image_quality TEXT, lat REAL, long REAL, 
+      timestamp INETEGER)
+    ''';
+const String albumTableCreationStrg = '''
+    CREATE TABLE IF NOT EXISTS $albumDataTableName(uid TEXT PRIMARY KEY, 
+    photo_uids TEXT, title TEXT, thumb_uid TEXT)
+  ''';
+const String keyCrosstableCreationStrg = '''
+    CREATE TABLE IF NOT EXISTS $keyCrosstableName(photo_uid Text, album_uid Text)
+  ''';
+// const String databaseConfigString = 'PRAGMA foreign_keys = ON';
+
+late final Database? _database;
+
+class DbNotOpenException implements Exception {}
+
+class FieldsMissingException implements Exception {}
+
+class DuplicateKeyException implements Exception {}
+
+class KeyNotFoundException implements Exception {}
+
+class SqlFilter {
+  final String column;
+  final String operator;
+  final String value;
+  final String? comparator;
+
+  const SqlFilter(this.column, this.operator, this.value, {this.comparator});
+
+  @override
+  String toString() {
+    if (comparator == null) {
+      return '("$column" $operator $value)';
+    } else {
+      return ' $comparator ("$column" $operator $value)';
+    }
+  }
+}
+
+Future<Database> _initDb() async {
+  try {
+    final dbPath = (await getApplicationDocumentsDirectory()).path;
+    return openDatabase(
+      join(dbPath, 'meta_data_db.db'),
+      onCreate: (db, version) async {
+        await _createPhotoDataTable(db);
+        await _createAlbumDataTable(db);
+      },
+      version: 1,
+      // onConfigure: (db) async {
+      //   await db.execute(databaseConfigString);
+      // },
+    );
+  } on MissingPlatformDirectoryException {
+    throw MissingPlatformDirectoryException(
+      "Could not get platform directory to create database",
+    );
```

Comment:
This is a bit redundant. If it throws an `MissingPlatformDirectoryException` you catch that error to throw the same error again.